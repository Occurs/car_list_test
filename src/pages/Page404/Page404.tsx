import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import Typography from "@material-ui/core/Typography";
import Grid from "@material-ui/core/Grid";
import Box from "@material-ui/core/Box";

const useStyles = makeStyles({
  container: {
    minHeight: 'inherit',
  }
});

const Page404 = () => {
  const classes = useStyles();
  return (
    <Grid className={classes.container} container direction="column" justify="center" alignItems="center">
      <Grid item>
        <Typography>
          <Box component="span" fontSize="32px" fontWeight="bold">
            404 - Not Found
          </Box>
        </Typography>
      </Grid>
      <Grid item>
        <Typography>
          <Box component="span" fontSize="18px">
            Sorry, the page you are looking for does not exist.
          </Box>
        </Typography>
      </Grid>
      <Grid item>
        <Typography>
          <Box component="span" fontSize="18px">
            You can always go back to the homepage.
          </Box>
        </Typography>
      </Grid>
    </Grid>
    // </Box>
  );
};

export default Page404;
