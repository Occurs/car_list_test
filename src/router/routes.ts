const routes = {
  carsList: '/',
  carCard: '/car/:stockNumber',
  favorites: '/favorites'
};

export default routes;