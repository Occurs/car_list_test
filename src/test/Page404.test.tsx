import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from 'react-router-dom';
import Page404 from "pages/Page404/Page404";
import routes from "router/routes";

describe("Page404 Component", () => {
  test("snapshot", () => {
    const component = renderer.create(
      <MemoryRouter initialEntries={['error-route']}>
        <Page404 />
      </MemoryRouter>
      );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
