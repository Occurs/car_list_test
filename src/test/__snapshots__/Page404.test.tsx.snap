// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page404 Component snapshot 1`] = `
<div
  className="MuiGrid-root makeStyles-container-1 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <div
      className="MuiBox-root MuiBox-root-3"
    >
      <img
        alt="logo"
        className="makeStyles-image-2"
        src="https://auto1-js-task-api--mufasa71.repl.co/images/logo.png"
      />
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      <span
        className="MuiBox-root MuiBox-root-4"
      >
        404 - Not Found
      </span>
    </p>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      <span
        className="MuiBox-root MuiBox-root-5"
      >
        Sorry, the page you are looking for does not exist.
      </span>
    </p>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      <span
        className="MuiBox-root MuiBox-root-6"
      >
        You can always go back to the 
        <a
          aria-current={null}
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
          href="/"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
        >
          <span
            className="MuiBox-root MuiBox-root-7"
          >
            homepage
          </span>
        </a>
        .
      </span>
    </p>
  </div>
</div>
`;
